File: platform/src/main/java/org/syncloud/platform/ssh/model/IdentifiedEndpoint.java
package org.syncloud.platform.ssh.model;

import com.google.common.base.Optional;

import java.io.Serializable;

public class IdentifiedEndpoint implements Serializable {
    private Endpoint enpoint;
    private Optional<Identification> id;

    public IdentifiedEndpoint(Endpoint endpoint, Optional<Identification> id) {
        this.enpoint = endpoint;
        this.id = id;
    }

    public Endpoint endpoint() {
        return enpoint;
    }

    public Optional<Identification> id() {
        return id;
    }
}

File: syncloud/src/main/java/org/syncloud/android/core/platform/model/IdentifiedEndpoint.kt
package org.syncloud.android.core.platform.model

import com.google.common.base.Optional
import org.syncloud.android.core.platform.model.Identification
import java.io.Serializable

class IdentifiedEndpoint(private val enpoint: Endpoint, private val id: Optional<Identification>) :
    Serializable {
    fun endpoint(): Endpoint {
        return enpoint
    }

    fun id(): Optional<Identification> {
        return id
    }
}